File: WssCRM/ClientApp/src/app/Models/Stage.ts
import {Point} from './Point'
export class Stage {

  constructor(
    public name?: string,
    public points?: Point[],
    public id?: number,
    public deleted?: boolean,
    public agreementStage?: boolean,
    public preAgreementStage?: boolean,
    public incomeStage? : boolean

  ) {
    this.points = [];

    this.deleted = false;
    this.agreementStage = false;
    this.incomeStage = false;
    this.preAgreementStage = false;
  }
}

File: WssCRM/Models/Stage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WssCRM.Models
{
    public class Stage
    {
        public string name;
        public int id;
        public bool deleted { get; set; }
        public List<Point> points = new List<Point>();
        public bool agreementStage { get; set; }
        public bool preAgreementStage { get; set; }
        public bool incomeStage { get; set; }
        public int Num { get; set; }
        public Stage()
        {

        }
        public Stage(string name, int id, bool deleted)
        {
            this.name = name;
            this.id = id;
            this.deleted = deleted;
        }
    }
}
